hw/drivers/chg_ctrl/da1469x_charger/src/da1469x_charger.c

```diff
@@ -0,0 +1,513 @@
+    SET_IF_VALID(CV_CHARGE_TIMER, cv_charge_timer);
+    SET_IF_VALID(JEITA_V_CHARGE, jeita_v_charge);
+    SET_IF_VALID(JEITA_V_PRECHARGE, jeita_v_precharge);
+    SET_IF_VALID(JEITA_V_REPLENISH, jeita_v_replenish);
+    SET_IF_VALID(JEITA_V_OVP, jeita_v_ovp);
+    SET_IF_VALID(JEITA_CURRENT, jeita_current);
+    SET_IF_VALID(VBAT_COMP_TIMER, vbat_comp_timer);
+    SET_IF_VALID(VOVP_COMP_TIMER, vovp_comp_timer);
+    SET_IF_VALID(TDIE_COMP_TIMER, tdie_comp_timer);
+    SET_IF_VALID(TBAT_MON_TIMER, tbat_mon_timer);
+    SET_IF_VALID(TBAT_COMP_TIMER, tbat_comp_timer);
+    SET_IF_VALID(THOT_COMP_TIMER, thot_comp_timer);
+    SET_IF_VALID(PWR_UP_TIMER, pwr_up_timer);
+
+    return 0;
+}
+
+#if MYNEWT_VAL(DA1469X_CHARGER_USE_CHARGE_CONTROL)
+static void
+da1469x_vbus_state_changed_event_cb(struct os_event *ev)
+{
+    assert(ev);
+    NVIC_SetPendingIRQ(CHARGER_STATE_IRQn);
+}
+
+static struct os_event da1469x_vbus_state_changed_event = {
+    .ev_cb = da1469x_vbus_state_changed_event_cb,
+};
+
+void
+VBUS_IRQHandler(void)
+{
+    CRG_TOP->VBUS_IRQ_CLEAR_REG = 1;
+    os_eventq_put(os_eventq_dflt_get(), &da1469x_vbus_state_changed_event);
+}
+
+void da1469x_usb_init(void)
+{
+    NVIC_SetVector(VBUS_IRQn, (uint32_t)VBUS_IRQHandler);
+    CRG_TOP->VBUS_IRQ_CLEAR_REG = 1;
+    NVIC_ClearPendingIRQ(VBUS_IRQn);
+    CRG_TOP->VBUS_IRQ_MASK_REG = CRG_TOP_VBUS_IRQ_MASK_REG_VBUS_IRQ_EN_FALL_Msk |
+                                 CRG_TOP_VBUS_IRQ_MASK_REG_VBUS_IRQ_EN_RISE_Msk;
+    NVIC_EnableIRQ(VBUS_IRQn);
+}
+
+/**
+* DA1469x interrupt handler CB
+* gets interrupt status and prints to console
+*/
+static void
+da1469x_charger_state_changed_cb(struct os_event *ev)
+{
+    struct da1469x_charger_dev *dev;
+    uint32_t irq_state;
+
+    assert(ev);
+    dev = ev->ev_arg;
+
+    irq_state = CHARGER->CHARGER_STATE_IRQ_STATUS_REG;
+    CHARGER->CHARGER_STATE_IRQ_CLR_REG = irq_state;
+    /* Interrupt generated by charger triggers out of schedule read */
+    charge_control_read(&dev->chg_ctrl, CHARGE_CONTROL_TYPE_STATUS,
+                        NULL, NULL, OS_TIMEOUT_NEVER);
+}
+
+static struct os_event da1469x_charger_state_changed_event = {
+    .ev_cb = da1469x_charger_state_changed_cb,
+};
+
+void
+CHARGER_STATE_IRQHandler(void)
+{
+    os_eventq_put(os_eventq_dflt_get(), &da1469x_charger_state_changed_event);
+}
+
+static void
+da1469x_charger_error_cb(struct os_event *ev)
+{
+    struct da1469x_charger_dev *dev;
+    uint32_t irq_state;
+
+    assert(ev);
+    dev = ev->ev_arg;
+
+    irq_state = CHARGER->CHARGER_ERROR_IRQ_STATUS_REG;
+    CHARGER->CHARGER_ERROR_IRQ_CLR_REG = irq_state;
+
+    /* Interrupt generated by charger triggers out of schedule read */
+    charge_control_read(&dev->chg_ctrl, CHARGE_CONTROL_TYPE_FAULT,
+                        NULL, NULL, OS_TIMEOUT_NEVER);
+}
+
+static struct os_event da1469x_charger_error_event = {
+    .ev_cb = da1469x_charger_error_cb,
+};
+
+void
+CHARGER_ERROR_IRQHandler(void)
```

Comment:
I would use small caps name here; this feels like a name of a macro/ value of an enum.